src/main/java/com/minecolonies/coremod/entity/ai/citizen/CitizenAI.java

```diff
@@ -0,0 +1,264 @@
+    }
+
+    private IState decideAiTask()
+    {
+        IState next = calculateNextState();
+        if (next == null || next == lastState && citizen.getCitizenAI().getState() != CitizenAIState.IDLE)
+        {
+            return null;
+        }
+
+        citizen.getCitizenData().setVisibleStatus(null);
+        Log.getLogger().warn(lastState+" -> "+next);
+        lastState = next;
+        return lastState;
+    }
+
+    private IState calculateNextState()
+    {
+        if (citizen.getCitizenJobHandler().getColonyJob() instanceof AbstractJobGuard)
+        {
+            if (shouldEat())
+            {
+                return CitizenAIState.EATING;
+            }
+            return CitizenAIState.WORK;
+        }
+
+        // Sick at hospital
+        if (citizen.getCitizenDiseaseHandler().isSick() && citizen.getCitizenDiseaseHandler().sleepsAtHospital())
+        {
+            citizen.getCitizenData().setVisibleStatus(VisibleCitizenStatus.SICK);
+            return CitizenAIState.SICK;
+        }
+
+        // Raiding
+        if (citizen.getCitizenColonyHandler().getColony().getRaiderManager().isRaided() && !(citizen.getCitizenJobHandler().getColonyJob() instanceof AbstractJobGuard))
+        {
+            citizen.getCitizenData().triggerInteraction(new StandardInteraction(Component.translatable(COM_MINECOLONIES_COREMOD_ENTITY_CITIZEN_RAID), ChatPriority.IMPORTANT));
+            citizen.setVisibleStatusIfNone(RAIDED);
+            return CitizenAIState.SLEEP;
+        }
+
+        // Sleeping
+        if (!WorldUtil.isPastTime(CompatibilityUtils.getWorldFromCitizen(citizen), NIGHT - 2000))
+        {
+            if (lastState == CitizenAIState.SLEEP)
+            {
+                citizen.setVisibleStatusIfNone(SLEEP);
+                citizen.getCitizenAI().setCurrentDelay(20 * 15);
+                return CitizenAIState.SLEEP;
+            }
+
+            if (citizen.getCitizenSleepHandler().shouldGoSleep())
+            {
+                citizen.getCitizenData().onGoSleep();
+                return CitizenAIState.SLEEP;
+            }
+        }
+        else
+        {
+            if (citizen.getCitizenSleepHandler().isAsleep() && !citizen.getCitizenDiseaseHandler().isSick())
+            {
+                citizen.getCitizenSleepHandler().onWakeUp();
+            }
+        }
+
+        // Sick
+        if (citizen.getCitizenDiseaseHandler().isSick() || citizen.getCitizenDiseaseHandler().isHurt())
+        {
+            citizen.getCitizenData().setVisibleStatus(VisibleCitizenStatus.SICK);
+            return CitizenAIState.SICK;
+        }
+
+        // Eating
+        if (shouldEat())
+        {
+            return CitizenAIState.EATING;
+        }
+
+        // Mourning
+        if (citizen.getCitizenData().getCitizenMournHandler().isMourning() && citizen.getCitizenData().getCitizenMournHandler().shouldMourn())
+        {
+            if (!citizen.getCitizenColonyHandler().getColony().getRaiderManager().isRaided())
```

Comment:
Ehm, not sure about this? What is this for? Will guards mourn during a raid btw?

Reply:
At the moments, guards don't mourn at all

Reply:
ye this is similar to the previous logic, guards are excluded from all this at the top

Reply:
But can this ever reach this in a raid at all?

Reply:
you're right raided doesnt reach here